ray-operator/controllers/ray/common/service_test.go

```diff
@@ -450,6 +450,49 @@ func TestBuildServiceForHeadPodPortsOrder(t *testing.T) {
 	}
 }
 
+func TestBuildHeadlessServiceForRayCluster(t *testing.T) {
+	svc := BuildHeadlessServiceForRayCluster(*instanceForSvc)
+
+	actualSelector := svc.Spec.Selector[utils.RayClusterLabelKey]
+	expectedSelector := instanceForSvc.Name
+	if !reflect.DeepEqual(expectedSelector, actualSelector) {
```

Comment:
Might be easier to just construct the expected Service object and do a `reflect.DeepEqual` check against that and the returned object from BuildHeadlessServiceForRayCluster. 

Maybe for a separate PR because this is just following the pattern of the other tests.